electrum/gui/qt/address_list.py

```diff
@@ -255,13 +255,16 @@ def refresh_row(self, key, row):
         if self.should_show_fiat():
             rate = fx.exchange_rate()
             fiat_balance_str = fx.value_str(balance, rate)
+            fiat_value = fx.fiat_value(balance, rate)
         else:
             fiat_balance_str = ''
+            fiat_value = None
         address_item = [self.std_model.item(row, col) for col in self.Columns]
         address_item[self.Columns.LABEL].setText(label)
         address_item[self.Columns.COIN_BALANCE].setText(balance_text)
         address_item[self.Columns.COIN_BALANCE].setData(balance, self.ROLE_SORT_ORDER)
         address_item[self.Columns.FIAT_BALANCE].setText(fiat_balance_str)
+        address_item[self.Columns.FIAT_BALANCE].setData(fiat_value, self.ROLE_SORT_ORDER)
```

Comment:
`fiat_value` can be `Decimal('NaN')`, e.g. if the rate is unavailable because the FX provider is down or it banned your IP address (e.g. CoinGecko via Tor is not reachable sometimes)
In that case, trying to sort this column will blurt out thousands of errors. (`Decimal('NaN') < Decimal('NaN')`)

The FX rate does not matter anyway if you just want to sort, so let's just sort based on the btc values:
`address_item[self.Columns.FIAT_BALANCE].setData(balance, self.ROLE_SORT_ORDER)`

Reply:
Thank you for taking the time to review this change. I amended my commit with your suggested changes